pkg/utils/utils.go

```diff
@@ -240,3 +240,32 @@ func RemoveString(slice []string, s string) (result []string) {
 	}
 	return
 }
+
+func EnvValueEqual(cenv, denv []v1.EnvVar) bool {
+	var found bool
+	if len(cenv) != len(denv) {
+		return false
+	}
+	for _, celem := range cenv {
+		found = false
+		if celem.ValueFrom != nil {
```

Comment:
we want to also compare `ValueFrom` to `ValueFrom` if they're both available

Reply:
My original plan was to skip comparing ValueFrom following the advice from Rich.  I'm afraid if we compare ValueFrom, we are to use reflect.DeepEqual?  That'd return false even if they look identical...

> Rich Megginson wrote: 
> I think the operator should ignore env vars where the value is provided by the downward api or otherwise provided by kubernetes.  That is, it is meaningless to compare the NODE_NAME value in the operator with the one from the currently running daemonset.

Reply:
this was something we brought up during scrum today... it'd be possible for users to have a `Value` compared to a `ValueFrom` otherwise and we would need to revert it to the appropriate one otherwise. Same with a `ValueFrom` vs another `ValueFrom`... if someone changes the variable to be based on the namespace rather than the pod name, it could cause issues in the fluentd image so we'd want to just keep that as we set it.

Reply:
Ok - if we expect the `ValueFrom` value to be identical in the current and the desired, then we should compare them too - I had thought perhaps that the value in the current one would have been altered by Kubernetes, that is, that Kubernetes would replace it with the actual value, but I guess that only happens when the actual pod is created.
So yes - we should compare `ValueFrom` also - but I guess we can't use `DeepEqual` because it doesn't work, and we don't know why . . .